client.go

```diff
@@ -687,6 +693,11 @@ type HostClient struct {
 	//
 	// By default request write timeout is unlimited.
 	WriteTimeout time.Duration
+	
+	// Maximum duration for full handshake reading and writing.
+	//
+	// By default request handshake timeout is unlimited.
+	HandshakeTimeout time.Duration
```

Comment:
I'm not sure if this is a good name. For http it now is the dial timeout, for https it is the dial timeout + handshake timeout.

Reply:
Also it should be backwards compatible which means it should default to `WriteTimeout` which was used before.

Reply:
Yes, I believe so, I just used such a naming in my fork, but what do you think we should name it? Or how to distinguish handshake read+write timeout , from write timeout? Or we should treat fasthttp writetimeout and readtimeout not only to request read/write but to handshake read/write operations as well?

Reply:
I think we should just keep using `WriteTimeout` for this as doing the handshake is kinda part of writing the request.

Reply:
But then subtract time spent on handshake from writeTimeout ?

Reply:
That's maybe also not the right solution. Maybe we should call it DialTimeout instead cause that's more what it is. It is also the timeout passed to the dialer for the whole dial operation. And I think when `DialTimeout` is 0 it should default to `WriteTimeout` so current users don't suddenly loose their timeout when the update fasthttp.